docs/build/credentials.md

```diff
@@ -30,8 +30,95 @@ authenticate with your app (e.g., `username`, `api_key`).
 a `Raw JSON` Credential in OpenFn to define whatever credential inputs are
-require (e.g., `{"api_key": "ADD-your-let", "url": "add-url"}`).
+required. E.g.:
 
-Some systems (Salesforce, OpenMRS, DHIS2) require an instanceUrl, host, or
-ApiUrl. Leave off the final "/" in these Urls: `https://login.salesforce.com` or
-`http://demo.openmrs.org/openmrs` or `https://play.dhis2.org`.
+```json
+{
+  "api_key": "someSecretKey",
+  "base_ur": "https://example.com/api/v2"
+}
+```
+
+Note that some systems (Salesforce, OpenMRS, DHIS2) require an instanceUrl,
+host, or ApiUrl. While most adaptors will handle a "trailing slash" in a URL
+gracefully, when in doubt you should leave it off. For example:
+
+- prefer `https://login.salesforce.com` to `https://login.salesforce.com/`,
+- use `http://demo.openmrs.org/openmrs` instead of
+  `http://demo.openmrs.org/openmrs/`,
+- and write `https://play.dhis2.org` rather than `https://play.dhis2.org/`.
+
+### Use Oauth2 credentials
+
+If Oauth2 _clients_ have been configured on your OpenFn instance, you can use
+them create Oauth credentials. These all work the same way:
+
+1. First you pick an Oauth credential type from the "New Credential" interface.
+2. Then you give it a name.
+3. You optionally select additional "scopes" that you want to use. (See the
+   in-app link to third-party documentation on scopes, depending on what
+   application you're using.)
+4. And then click "Sign in with \_\_\_\_\_\_".
+
+You will be asked by the third-party application to verify your identity and
+that you'd like to use Oauth. Once you accept, back on OpenFn you'll be able to
+save and use your new credential just like any other.
+
+:::tip
+
+If you don't see an Oauth credential type available for your app, contact the
+superuser who deployed your instance of OpenFn and ask them to set up some Oauth
+clients for your app. Users of the OpenFn platform SaaS can post on
+[community.openfn.org](https://community.openfn.org) or send an email to
+[support@openfn.org](support@openfn.org).
+
+:::
+
+#### e.g., GoogleSheets Oauth Credential
+
+Note the credential selects only required scopes for Google Sheets.
+
+![Google Oauth](/img/google-oauth2.png)
+
+#### e.g., Salesforce Oauth Credential
+
+Note that you can choose which scopes to access on Salesforce.
+
+![Salseforce Oauth](/img/salesforce-oauth2.png)
+
+:::tip
+
+Check out the dedicated [Adaptor docs](/adaptors) page for app-specific
+credential guidance.
+
+:::
+
+### Creating a dedicated "integration user" for your OpenFn workflow
+
+To keep target systems as secure and controlled as possible, we recommend that
+credentials used in the integration be granted API-only access to the target
+application. You _can_ use your personal user as an OpenFn credential for your
+workflow, but we recommend that you create a dedicated "OpenFn" integration user
+or service account user to access your target applications.
+
```

Comment:
Should we add an example for systems that provides API only user ?
Eg:
```
Salesforce: You can create API-only users to perform automated tasks and integrations without requiring full user access.
```

Reply:
addressed in next commit